ops/model.py

```diff
@@ -996,10 +997,15 @@ def relation_get(self, relation_id, member_name, is_app):
         if not isinstance(is_app, bool):
             raise TypeError('is_app parameter to relation_get must be a boolean')
 
+        if is_app and not JujuVersion.from_environ().has_app_data():
+            return {}
+
+        args = ['relation-get', '-r', str(relation_id), '-', member_name]
+        if is_app:
+            args.append('--app')
```

Comment:
I'm happy to use this logic always rather than the '--app=False' logic. It makes it easier to support 2.6.* if-and-when for anything that doesn't need app data.